### PR TITLE
feat: Radio buttons for text and drawable.

### LINKED ISSUE #114: Changing Level or Mode when Preview Badge is displaying should take effect without having to click Preview Button to apply change
**Describe the bug**

When text has been enter or image selected  and preview button clicked and preview Badge display active,
if Mode is change , preview Badge mode should change dynamically and also if speed is change Preview Badge speed should also be change dynamically without having to click the preview button to apply changes.

**To Reproduce**

Enter a text or select an image
click on preview
change level or mode (but preview Badge Configuration remains same)

**Expected behavior**
if Mode is change , preview badge mode should change dynamically and also if speed is change Preview Badge speed should also be affected dynamically without having to click the preview button for change to take place.

|               | |
|---------------|-|
|Device         |GIONEE M5|
|Android Version|LOLLIPOP 5.1  |

**Additional context**

**Would you like to work on the issue?**
I will very much like to work on this issue

### LINKED ISSUE #136: There should be an option to clear the preview.
**Describe the feature you'd like**
Currently, there is no option to clear the preview from the preview badge once preview is clicked. It can be changed but not cleared.
There should be a button which clears the preview on being clicked.

**Would you like to work on the issue?**

YES.

### LINKED ISSUE #151: Different Sections for Text and Drawable's inside the app.
**Describe the feature you'd like**
Currently, both the text and drawable are in the same screen. Different fragments will allow the user to have a better experience when selecting a text or a drawable.

**Additional context**
This will solve the issue of not being able to send the selected drawable to the LED Badge 

**Would you like to work on the issue?**
Yep, Im in

### LINKED ISSUE #168: Make badge preview fixes
**Describe the feature you'd like**
make badge preview fixed and other items scroll-able only when the screen is in portrait mode.

**Additional context**
It will allow user to see the preview all the time.

**Would you like to work on the issue?**
YES

### LINKED ISSUE #169: Make only one Preview button
**Describe the feature you'd like**
Only one preview button will make the app more user friendly while having two preview button doesn't look good for now.

**Additional context**
The preview button will be besides the current Send button.

**Would you like to work on the issue?**
YES

### LINKED ISSUE #170: Either text or drawable must be enabled
**Describe the bug**
Currently when in portrait mode, both can be selected but when we write down text and preview it the drawable highlight must go away.

**To Reproduce**
1. Open app
2. click on any drawable and preview it.
3. Type any text and preview it

Here still the drawable will be highlighted.

**Expected behavior**
The drawable must not be highlighted once the text is previewed.

**Logs**
N/A

**Screenshots**
Screenrecorder-2019-04-06-19-12-42-321

**Smartphone Info:**

| Device 	| Android Version 	|
|---------	|:---------------:	|
| Redmi 4 	| Android 7.1.2 	|

**Would you like to work on the issue?**
YES

### PR DESCRIPTION
**Pull request by yashk2000**
Fixes #151 
Fixes #168 
Fixes #169
Fixes #170 
Fixes #136 
Fixes #114

Changes:
- Radio buttons have been added to choose between text and drawables.
- Preview button has been completely removed. The preview badge is continually updated as the user keeps changing data.

Screenshots/GIF for the change:

ezgif com-video-to-gif (18)

screencap

screencap1

### CODE REVIEW

**Comment by yashk2000** [comment]
@adityastic please review.

**Comment by adityastic** [comment]
This Fixes #168 #169 #170 #136 #114, Add them @yashk2000

**Comment by adityastic** [comment]
@yashk2000 Let me work on the sending logic, Ive already started working on it. It will allow you to send any drawable, not only the ones present in the app. Please share your feedback with this request

**Comment by yashk2000** [comment]
@adityastic ok. I will remove the sending logic. You can continue working on it.

**Comment by adityastic** [comment]
> @adityastic ok. I will remove the sending logic. You can continue working on it.

Thanks ;)

**Comment by yashk2000** [comment]
@adityastic done :+1:

**Comment by adityastic** [comment]
Let me review it :)

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt [hunk 1]
```diff
@@ -72,6 +100,26 @@ class MessageActivity : AppCompatActivity() {
         speed.adapter = ArrayAdapter<String>(this, spinnerItem, Speed.values().mapIndexed { index, _ -> (index + 1).toString() })
         mode.adapter = ArrayAdapter<String>(this, spinnerItem, Mode.values().map { getString(it.stringResId) })
 
+        val listOfDrawables = listOf(
```
Comment (adityastic) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: Not used anywhere

File: app/src/main/java/com/nilhcem/blenamebadge/adapter/DrawableAdapter.kt [hunk 2]
```diff
@@ -30,16 +31,32 @@ class DrawableAdapter(private val context: Context, private val list: List<Drawa
...
+        return selectedPosition
+    }
+
+    fun setListener(listener: OnDrawableSelected) {
+        this.listener = listener
+    }
 
     override fun getItemCount() = list.size
 
+    fun resetSelected() {
```
Comment (adityastic) [inline app/src/main/java/com/nilhcem/blenamebadge/adapter/DrawableAdapter.kt]: this function is never used

File: app/src/main/java/com/nilhcem/blenamebadge/adapter/DrawableAdapter.kt [hunk 3]
```diff
@@ -30,16 +31,32 @@ class DrawableAdapter(private val context: Context, private val list: List<Drawa
         return if (selectedPosition == -1) null else list[selectedPosition]
     }
 
-    fun getDefaultItem(): DrawableInfo = DrawableInfo(context.resources.getDrawable(R.drawable.mix2))
+    fun getSelectedItemIndex(): Int {
```
Comment (adityastic) [inline app/src/main/java/com/nilhcem/blenamebadge/adapter/DrawableAdapter.kt]: remove this function too

**Comment by iamareebjamal** [comment]
Radio Buttons should be side by side, and once a radio button is selected, the layout for other should be hidden

**Comment by yashk2000** [comment]
@adityastic I removed the redundant code. @iamareebjamal I will make those changes.

**Comment by yashk2000** [comment]
screencap

screencap1

**Comment by yashk2000** [comment]
@adityastic @iamareebjamal please review. I have made the changes.

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt [hunk 4]
```diff
@@ -53,20 +59,35 @@ class MessageActivity : AppCompatActivity() {
...
+
+        override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+            selectText()
+        }
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.message_activity)
+        radioText.isChecked = true
```
Comment (adityastic) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]:
```suggestion
```

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt [hunk 5]
```diff
@@ -101,55 +124,79 @@ class MessageActivity : AppCompatActivity() {
...
+                        textSection.setVisibility(View.GONE)
+                        selectDrawable(drawableRecyclerAdapter.getSelectedItem())
+
+                        removeListeners()
+                        drawableRecyclerAdapter.setListener(object : OnDrawableSelected {
+                            override fun onSelected(selectedItem: DrawableInfo?) {
+                                selectDrawable(selectedItem)
+                            }
+                        })
+                    }
```
Comment (adityastic) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]:
```suggestion
                    }
                    radioText.isChecked = true
```

**Comment by adityastic** [comment]
@iamareebjamal Peer Review done

**Comment by adityastic** [comment]
Please add #151 to the list

**Comment by yashk2000** [comment]
@iamareebjamal please review.

**Pull request merged**